community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -242,6 +242,22 @@ public Status getReasonIfTerminated()
         return terminationReason;
     }
 
+    void markForTermination( long expectedReuseCount, Status reason )
+    {
+        terminationReleaseLock.lock();
+        try
+        {
+            if ( expectedReuseCount == reuseCount )
+            {
+                markForTermination( reason );
```

Comment:
common part from `markForTermination` should be extracted and invoked here instead to avoid double locking